pkg/io/fileWriter_test.go

```diff
@@ -0,0 +1,38 @@
+package io
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMakeDirForFile_HappyPath(t *testing.T) {
+	filePath, err := ioutil.TempDir("", "test")
+	require.NoError(t, err)
+
+	err = MakeDirForFile(filePath, "test")
```

Comment:
You need to create something inside `filePath` now, because `filePath` is now a directory (and not a `file`, BTW) and it does exist already (because that's the way `TempDir` works).